nebula-logger/core/tests/logger-engine/classes/Logger_Tests.cls

```diff
@@ -166,27 +166,133 @@ private class Logger_Tests {
+            System.assertEquals(firstMockScenario, logEntryEvent.TransactionScenario__c);
+        }
     }
 
     @IsTest
-    static void it_should_use_the_last_specified_scenario_when_parameter_is_false() {
+    static void it_should_use_the_last_specified_scenario_as_transaction_scenario_when_parameter_is_false() {
+        LoggerDataStore.setMock(LoggerMockDataStore.getEventBus());
         LoggerParameter.setMock(new LoggerParameter__mdt(DeveloperName = 'UseFirstSpecifiedScenario', Value__c = String.valueOf(false)));
         String firstMockScenario = 'the first test scenario specified for this transaction';
         String secondMockScenario = 'the second test scenario specified for this transaction';
 
         Logger.setScenario(firstMockScenario);
+        Logger.info('hello');
+        Logger.setScenario(secondMockScenario);
+        Logger.info('world');
+        Logger.saveLog();
+
+        System.assertEquals(2, LoggerMockDataStore.getEventBus().getPublishedPlatformEvents().size());
+        for (LogEntryEvent__e logEntryEvent : (List<LogEntryEvent__e>) LoggerMockDataStore.getEventBus().getPublishedPlatformEvents()) {
+            System.assertEquals(secondMockScenario, logEntryEvent.TransactionScenario__c, 'The last specified scenario should have been used');
+        }
+    }
+
+    @IsTest
+    static void it_should_set_transaction_scenario_and_entry_scenario_for_published_log_entry_events() {
+        LoggerDataStore.setMock(LoggerMockDataStore.getEventBus());
+        String firstMockScenario = 'the first test scenario specified for this transaction';
+        String secondMockScenario = 'the second test scenario specified for this transaction';
+
+        Logger.setScenario(firstMockScenario);
+        Logger.info('hello');
         Logger.setScenario(secondMockScenario);
+        Logger.info('world');
+        Logger.saveLog();
 
-        System.assertEquals(secondMockScenario, Logger.getScenario(), 'The last specified scenario should have been used');
+        System.assertEquals(2, LoggerMockDataStore.getEventBus().getPublishedPlatformEvents().size());
+        List<LogEntryEvent__e> publishedLogEntryEvents = (List<LogEntryEvent__e>) LoggerMockDataStore.getEventBus().getPublishedPlatformEvents();
+        LogEntryEvent__e firstLogEntryEvent = publishedLogEntryEvents.get(0);
+        System.assertEquals(firstMockScenario, firstLogEntryEvent.EntryScenario__c);
+        System.assertEquals(firstMockScenario, firstLogEntryEvent.TransactionScenario__c);
+        LogEntryEvent__e secondLogEntryEvent = publishedLogEntryEvents.get(1);
+        System.assertEquals(secondMockScenario, secondLogEntryEvent.EntryScenario__c);
+        System.assertEquals(firstMockScenario, secondLogEntryEvent.TransactionScenario__c);
+    }
+
+    @IsTest
+    static void it_should_end_scenario_when_no_scenario_was_previously_set() {
```

Comment:
ha, love this one

Reply:
Some of these method names were fun to write